common/common.go

```diff
@@ -486,3 +487,30 @@ func GetExitStatus(err error) (int, error) {
 	}
 	return -1, err
 }
+
+// reference to GOARM, needs to be globally, if in GetArch() function it resolves to zero
+//go:linkname goarm runtime.goarm
+var goarm uint8
+
+// GetArch returns the current architecture. It returns runtime.GOARCH if not on a arm device
+// and armv6l, armv7l, arm64 otherwise
+func GetArch() string {
+	arch := runtime.GOARCH
+	switch arch {
```

Comment:
it should be safe to do this appc lookup for all architectures, not just for arm.